lab_3_generate_by_ngrams/main.py

```diff
@@ -41,6 +43,21 @@ def _tokenize(self, text: str) -> Optional[tuple[str, ...]]:
         In case of corrupt input arguments, None is returned.
         In case any of methods used return None, None is returned.
         """
+        if not isinstance(text, str) or not text:
+            return None
+
+        tokens = []
+        for token in text.lower():
+            if token == ' ' and tokens[-1] != self._end_of_word_token:
```

Comment:
`token == ' '` better change to `token` - it is a more pythonic way

Reply:
I don't understand you... How is this an equivalent of space check? Can you elaborate on it?

Reply:
Ok, got it